Hestia/app/src/main/java/hestia/UI/SlideDialog.java

```diff
@@ -1,126 +1,139 @@
-//            @Override
-//            public void onClick(View view) {
-//                //send data to the server
-//                dismiss();
-//            }
-//        });
+    private ArrayList<Activator> fields;
+    private Context context;
+    private ClientInteractionController cic;
+
+    public SlideDialog(Context a, ArrayList<Activator> fields, Device d) {
+        super(a);
+        this.context = a;
+        this.cic = ClientInteractionController.getInstance();
+        this.fields = fields;
+        this.d = d;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        setContentView(R.layout.slide_dialog);
+
+        final LinearLayout lm = (LinearLayout) findViewById(R.id.linearMain);
+        int count = 0;
+
+        for (Activator activator : fields) {
+            LinearLayout ll = new LinearLayout(context);
+
+            // Add text
+            TextView name = new TextView(context);
+            name.setText(activator.getName());
+            ll.addView(name);
+
+            //Add field
+            Float currState = Float.parseFloat(activator.getState().toString());
+            SeekBar bar = createSeekBar(currState ,count, activator);
+            ll.addView(bar);
+
+            lm.addView(ll);
+            count++;
+        }
+        LinearLayout ll = generateButtons();
+        lm.addView(ll);
+    }
+
+
+    private SeekBar createSeekBar(float progress, int count, Activator a){
+        final Activator act = a;
+        SeekBar bar = new SeekBar(context);
+        final int max_int = Integer.MAX_VALUE;
+        bar.setMax(max_int);
+        bar.setProgress((int)progress* max_int);
+        bar.setLayoutParams(new LinearLayout.LayoutParams(800,80));
+        bar.setId(count);
+        bar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+
+            }
+
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+
+            }
+
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+                float value = (float)seekBar.getProgress()/max_int;
+                ActivatorState<Float> state = act.getState();
+                state.setState(value);
+                cic.setActivatorState(d,act.getId(),state);
+            }
+        });
+        return bar;
+    }
+
+
+    @Override
+    public void onClick(View v) {
+
+        switch (v.getId()) {
+            case 11:
+                Toast.makeText(getContext(), "Leaving", Toast.LENGTH_SHORT).show();
+                dismiss();
+                break;
+            case 12:
+                Toast.makeText(getContext(), "Cancel", Toast.LENGTH_SHORT).show();
+                dismiss();
+                break;
+            default:
```

Comment:
What does 11 and 12 represent?
Clean code recommends against using magic numbers.